src/util/createInstantSearch.js

```diff
@@ -75,19 +143,40 @@ export const createInstantSearch = instantSearchOptions => {
       warn('The result of `getState()` needs to be passed to `hydrate()`.');
       return;
     }
-    const { lastResults } = instantSearchState;
-    search.searchParameters = lastResults._state;
+    const { lastResults, derived } = instantSearchState;
     search.helper = algoliaHelper(searchClient, indexName, lastResults._state);
     search.helper.lastResults = new SearchResults(
       new SearchParameters(lastResults._state),
       lastResults._rawResults
     );
 
+    // TODO: this is still causing _two_ close to identical requests on the frontend
```

Comment:
the requests could come from somewhere else, but a main difference is that the second request contains the ais-highlight tags everywhere, while the first doesn't.